AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/environments/EnvironmentsRest.java

```diff
@@ -42,4 +43,13 @@ public List<EnvironmentDTO> getEnvironments(@ApiParam("Also returns Environment
         return environments;
     }
 
+    @GET
+    @Path("/contexts")
+    @ApiOperation(value = "Get all contexts", notes = "Returns all contexts as environments")
+    public List<EnvironmentDTO> getContexts() {
+        List<EnvironmentDTO> environments = new ArrayList<>();
+        contextLocator.getAllEnvironments().stream().map(EnvironmentDTO::new).forEach(environments::add);
+        return environments;
```

Comment:
You don't need the `forEach` to get from the stream to the list. You can use `stream().collect(Collectors.toList())`

Your implementation would look like this:
```
public List<EnvironmentDTO> getContexts() {
    return contextLocator.getAllEnvironments().stream().map(EnvironmentDTO::new).collect(Collectors.toList());
}
```

This eliminates the mutation (bad) of the temporary variable (code smell!) `environments`..